File: facade-design/computer.py
from cpu import CPU
from memory import Memory
from gpu import GPU
class Computer:
    def __init__(self):
        self.cpu = CPU()
        self.memory = Memory()
        self.gpu = GPU()

    def start(self):
        self.cpu.freeze()
        self.memory.load()
        self.cpu.jump(23)
        self.cpu.execute("load image")
        self.gpu.render()

File: facade/Computer.java
package facade;

public class Computer {
    private CPU cpu;
    private GPU gpu;
    private Memory memory;
    public Computer() {
        cpu = new CPU();
        gpu = new GPU();
        memory = new Memory();
    }

    public void start() {
        this.cpu.freeze();
        this.memory.load();
        this.cpu.jump(23);
        this.cpu.execute("load doom");
        this.gpu.render();
    }
}
